pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -260,7 +258,8 @@ def format_checkpoint_name(self, epoch, metrics, ver=None):
         filename = self._format_checkpoint_name(self.filename, epoch, metrics, prefix=self.prefix)
         if ver is not None:
             filename = self.CHECKPOINT_JOIN_CHAR.join((filename, f'v{ver}'))
-        return os.path.join(self.dirpath, f'{filename}.ckpt')
+        ckpt_name = f'{filename}.ckpt'
+        return os.path.join(self.dirpath, ckpt_name) if self.dirpath else ckpt_name
```

Comment:
here `self.dirpath` can't be None right? if it's none in init it will be assigned to `default_root_dir` from Trainer.

Reply:
Should not be None during normal usage, however, it is in this case (covered by a test):
https://github.com/PyTorchLightning/pytorch-lightning/blob/2d17626f902cd7822b3d8539acb919a240203ffd/tests/callbacks/test_model_checkpoint.py#L135